payroll/models/hr_payslip.py

```diff
@@ -551,7 +540,7 @@ def _compute_payslip_line(self, rule, localdict, lines_dict):
         previous_amount = rule.code in localdict and localdict[rule.code] or 0.0
         # compute the rule to get some values for the payslip line
         values = rule._compute_rule(localdict)
-        key = rule.code + "-" + str(localdict["contract"].id)
+        key = (rule.code or str(rule.id)) + "-" + str(localdict["contract"].id)
```

Comment:
@nimarosa If the rule has a code it is used. Otherwise it uses the id. I think this is a reasonable.